web/src/OverviewTable.tsx

```diff
@@ -478,29 +536,53 @@ function resourceTypeLabel(r: UIResource): string {
   return "Unknown"
 }
 
+function hasWidgets(view: Proto.webviewView): boolean {
+  return !!view.uiButtons?.length
+}
+
+function viewToRowValues(
+  view: Proto.webviewView,
+  logStore: LogStore
+): RowValues[] {
+  return (
+    view.uiResources?.map((r) =>
+      uiResourceToCell(r, view.uiButtons || [], logStore)
+    ) || []
+  )
+}
+
 export default function OverviewTable(props: OverviewTableProps) {
   let logStore = useLogStore()
-  const data = React.useMemo(
-    () =>
-      props.view.uiResources?.map((r) => uiResourceToCell(r, logStore)) || [],
-    [props.view.uiResources]
-  )
+  const data = React.useMemo(() => viewToRowValues(props.view, logStore), [
+    props.view.uiResources,
+    props.view.uiButtons,
+  ])
 
   const {
     getTableProps,
     getTableBodyProps,
     headerGroups,
     rows,
     prepareRow,
+    columns,
   } = useTable(
     {
-      columns,
+      columns: columnDefs,
       data,
       autoResetSortBy: false,
     },
     useSortBy
   )
 
+  // if there are no widgets, hide the widgets column
+  // TODO(matt) is this actually what we want?
```

Comment:
fwiw this seems reasonable to me

Reply:
yeah, it's at least fine. my main hesitations are:
1) it's inconsistent with other columns (e.g., we don't hide endpoints or podids when all of their cells are empty)
2) showing an empty column allows space for advertising the feature, though I'm not sure we'd do that at this point anyway